### ISSUE TITLE
MissingMethodException: set_Encoder on JsonSerializationOptions

### ISSUE THREAD

**Issue by gregsdennis** [dotnet/runtime#82798]
### Description

It seems that the `JsonSerializationOptions.Encoder` setter method is missing for some reason in the latest System.Text.Json nuget package.

### Reproduction Steps

```c#
[Test]
public void Test()
{
	var options = new JsonSerializerOptions
	{
		Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping
	};
}
```

### Expected behavior

It sets the `Encoder` property.

### Actual behavior

💥 exception

```
System.MissingMethodException : Method not found: 'Void System.Text.Json.JsonSerializerOptions.set_Encoder(System.Text.Encodings.Web.JavaScriptEncoder)'.
```

### Regression?

Setting this property works with v6 of the nuget package.  I haven't tried other v7s.

### Known Workarounds

None.

### Configuration

I'm running Win11 (kept up to date, currently `10.0.22621 Build 22621`) x64.

I've verified for .Net Core 3.1 and .Net 6 with System.Text.Json v7.0.2.

**Comment by gregsdennis** [comment]
☝️ I will never think its not funny that I open an issue and get tagged.

**Comment by jkotas** [comment]
Could you please share project and steps required to reproduce the problem?

This type of problem is typically caused by same assembly getting loaded multiple times.

**Comment by gregsdennis** [comment]
I can't seem to reproduce minimally, but it happens in the Json.More.Tests project in https://github.com/gregsdennis/json-everything.

The entire test suite for the solution runs fine for v6, but updating to v7 with no other changes yields the exception and a bunch of tests fail.

**Comment by eiriktsarpalis** [comment]
@gregsdennis can you share a branch of that repo with the projects updated to `net7.0`? Is there a specific test that is consistently reproducing the error?

**Comment by gregsdennis** [comment]
I've added `net7.0` to the test runner project.  I need to keep supporting `netstandard2.0` in the lib, though.

Branch: https://github.com/gregsdennis/json-everything/tree/stj-update
Test: https://github.com/gregsdennis/json-everything/blob/stj-update/Json.More.Tests/DevTest.cs

Interestingly, it works fine in .net 7, but still fails in .net 6 and .net core 3.1.

**Comment by eiriktsarpalis** [comment]
I can reproduce the error locally with the following command:
```bash
dotnet test -f net6.0 JsonSchema.OpenApi.Tests/ --filter 'Name~ConformanceTest'
```
Looking at the output folder I'm seeing copies for both `System.Text.Json.dll` and `System.Text.Encodings.Web.dll` (both version 7.0.0.0 targeting `net6.0` taken from the latest NuGet package). It also seems that these are the assemblies being loaded at runtime. The issue seems to go away as soon as I convert the test project to a console app that just runs the test method in hand, so I suspect NUnit might be related to this somehow.

**Comment by gregsdennis** [comment]
> I suspect NUnit might be related to this somehow

That's bothersome.  It would be quite an undertaking to convert all of my tests to XUnit.

I'll raise with NUnit.  Can we leave this open for now?

**Comment by gregsdennis** [comment]
Yeah, confirmed it's NUnit.  I had updated that from 4.3.1 to 4.4.0 just before.  Reverting that fixes the problem.

**Comment by gregsdennis** [comment]
Made a minimal example.

[TestProject1.zip](https://github.com/dotnet/runtime/files/10865559/TestProject1.zip)

**Comment by gregsdennis** [comment]
tl;dr of linked issue: update to NUnit v4.4.2.
